pkg/tsdb/prometheus/testdata/matrix_golden.txt

```diff
@@ -0,0 +1,137 @@
+🌟 This was machine generated.  Do not edit. 🌟
+
+Frame[0] {
+    "custom": {
+        "resultType": "matrix"
+    }
+}
+Name: every_0
+Dimensions: 2 Fields by 10 Rows
++-------------------------------+--------------------------+
+| Name: Time                    | Name: Value              |
+| Labels:                       | Labels: __name__=every_0 |
+| Type: []time.Time             | Type: []*float64         |
++-------------------------------+--------------------------+
+| 1970-01-01 00:00:00 +0000 UTC | 0                        |
+| 1970-01-01 00:00:01 +0000 UTC | 1                        |
+| 1970-01-01 00:00:02 +0000 UTC | 2                        |
+| 1970-01-01 00:00:03 +0000 UTC | 3                        |
+| 1970-01-01 00:00:04 +0000 UTC | 4                        |
+| 1970-01-01 00:00:05 +0000 UTC | 5                        |
+| 1970-01-01 00:00:06 +0000 UTC | 6                        |
+| 1970-01-01 00:00:07 +0000 UTC | 7                        |
+| 1970-01-01 00:00:08 +0000 UTC | 8                        |
+| 1970-01-01 00:00:09 +0000 UTC | 9                        |
++-------------------------------+--------------------------+
+
+
+
+Frame[1] {
+    "custom": {
+        "resultType": "matrix"
+    }
+}
+Name: every_1
+Dimensions: 2 Fields by 10 Rows
++-------------------------------+--------------------------+
+| Name: Time                    | Name: Value              |
+| Labels:                       | Labels: __name__=every_1 |
```

Comment:
This should either be:

```suggestion
Name: every_1
Dimensions: 2 Fields by 10 Rows
+-------------------------------+--------------------------+
| Name: Time                    | Name: Value              |
| Labels:                       | Labels:  |
```

OR (wide frame)
```suggestion
Name: 
Dimensions: 2 Fields by 10 Rows
+-------------------------------+--------------------------+
| Name: Time                    | Name: every_1              |
| Labels:                       | Labels: ... |
```

Reply:
good catch. i copied that over from the wide frame PR

Reply:
@ryantxu updated in ce89a87. just to confirm, only the `__name__` field should be filtered from labels on multi frame responses right?

Reply:
I agree with @ryantxu that this looks like correct behavior. However, in browser mode + previous server mode, we kept the  metric name (`__name__` label) in the list of labels. So this could change how results in some panels look/maybe break some transformations. I am wondering, if we shouldn't start with the list of v9 breaking changes and introduce this change in v9?

Reply:
Here is the example of different result of the same query on this branch (no __name__ label) and in browser mode/main branch:
![image](https://user-images.githubusercontent.com/30407135/148551761-24964538-7ee5-42b9-b4ed-b58043b09fe5.png)

Reply:
@ivanahuckova i'll revert in this PR so it can be handled separately. thanks for checking!

Reply:
reverted in f311655